src/components/Image.tsx

```diff
@@ -0,0 +1,27 @@
+import { useRef } from 'react';
+import { Animated, ImageProps as NativeProps } from 'react-native';
+
+export type ImageProps = Omit<NativeProps, 'source'> & {
+  source: string;
+};
+
+function Image({ source, style }: ImageProps) {
+  const opacity = useRef(new Animated.Value(0));
+
+  const onLoadEnd = () => {
+    Animated.spring(opacity.current, {
```

Comment:
Maybe timing here? spring will have it animate to 1, then back to 0.something for a bit, might look odd for opacity animation?